manifests/config.pp

```diff
@@ -12,6 +12,13 @@
     validate_cmd => "${dns::named_checkconf} %",
   }
 
+  file { $dns::logdir:
+    ensure => directory,
+    owner  => $dns::params::user,
+    group  => $dns::params::group,
+    mode   => '0755',
+  }
```

Comment:
This is only needed if there is a log channel with type file, right? Can we include it only if needed to keep the number of managed files down?

Reply:
I'm not sure how I would selectively ensure this directory if and only if a defined type was instantiated with specific options.